perses/rjmc/topology_proposal.py

```diff
@@ -2344,6 +2350,8 @@ def _constraint_repairs(atom_map, old_system, new_system, old_topology, new_topo
         """
         Given an adjusted atom map (corresponding to the true indices of the new: old atoms in their respective systems), iterate through all of the
         atoms in the map that are hydrogen and check if the constraint length changes; if so, we do not map.
+
+        Returns the new atom map and also the deleted atom indices.
```

Comment:
I know you didn't write the original docstring, but whenever we encounter a non-compliant docstring and have to work to figure out what the arguments are, we should fix the docstring.

The docstring should clearly document parameters and returns.

I'm not quite sure if the `{old|new}_{system|topology}` refer to the ENTIRE system (e.g. with solvent and protein) or one just representing the alchemical region---do you know?

It's also weird to me that we are passing a dict atom map, rather than an `AtomMap` object---we probably need to refactor that in future.

```
Parameters:
------------
atom_map : dict of int : int
    The atom mapping to repair; will not be modified.
    atom_map[old_index] refers to the new index of the corresponding atom in the new system
old_system : openmm.System
    The old OpenMM System containing parameters
new_system : openmm.System
    The old OpenMM System containing parameters
old_topology : openmm.app.Topology
    Old Topology
new_topology : openmm.app.Topology
    New Topology

Returns
--------
atom_map : dict of int : int 
    New atom map where some mapped atoms have been removed
deleted_new_atoms : list of int
    List of new_system atoms that were removed from the mapping
```
Also, why not return the deleted `old -> new` mappings as a `dict` rather than just the deleted new atoms?

Reply:
should be addressed now in commit https://github.com/choderalab/perses/pull/1125/commits/4f6b782b03c0e20d07c34a0c8c3d5176da7076fb